functions/src/aqi-calculation/cleaned-reading.ts

```diff
@@ -127,24 +136,43 @@ function getHourlyAverages(
       // https://cfpub.epa.gov/si/si_public_file_download.cfm?p_download_id=540979&Lab=CEMM
       // Expressed this way to avoid imprecision of floating point arithmetic.
       const MEASUREMENT_COUNT_THRESHOLD = 23;
+      const THREE_HOURS = 3;
       if (validReadings.length >= MEASUREMENT_COUNT_THRESHOLD) {
         averages[hoursAgo] = averageReadings(validReadings);
       } else if (nonNullReadings.length >= MEASUREMENT_COUNT_THRESHOLD) {
         // This case means that meanPercentThreshold was the final straw to make
-        // this hour lack enough valid readings
+        // this hour lack enough valid readings, since there were enough non-null
+        // readings but not enough readings with a low enough mean percent difference.
         averages[hoursAgo] = null;
-        // TODO: write invalid reason to sensor doc, or propagate
+
+        // We only propagate the error if the error occurred in the most recent
+        // three hours, since any null hour that occurred more than 3 hours ago
+        // can safely be discarded without affecting the validity of the AQI.
+        if (hoursAgo < THREE_HOURS) {
+          aqiCalculationErrors[
+            InvalidAqiErrors.NotEnoughRecentValidReadings
+          ] = true;
+        }
       } else {
         // In this case, not enough readings were received from PurpleAir
         averages[hoursAgo] = null;
-        // TODO: write invalid reason to sensor doc, or propagate
+
+        // We only propagate the error if the error occurred in the most recent
+        // three hours, since any null hour that occurred more than 3 hours ago
+        // can safely be discarded without affecting the validity of the AQI.
+        if (hoursAgo < THREE_HOURS) {
+          aqiCalculationErrors[InvalidAqiErrors.NotEnoughNewReadings] = true;
+        }
       }
     }
   } else {
     // If no buffer exists, there are no valid readings for any hour
     averages.fill(null);
+
+    // If there is no buffer, then there's not enough new readings
+    aqiCalculationErrors[InvalidAqiErrors.NotEnoughNewReadings] = true;
```

Comment:
Is this the right error for there not being a buffer? It seems like there'd only not be a buffer if the sensor was brand new (or just being re-activated). "Not enough new readings" is totally fine w/ me, it seems to get across the idea that it needs more readings to report AQI, but I also would consider adding the reason "needs more time to collect data" or something that gets across that the error exists b/c it's new.

Reply:
I could make a separate error for this, but the second time we try to calculate the AQI, it will be indistinguishable from "not enough new readings" since the buffer defaults to being filled with nulls. I can make the explanation for this error more clear though and add that "if you just activated or added this sensor, it may also need time to collect enough readings to calculate an AQI"